File: app/src/main/java/develop/y/zhzl/main/DarkViewActivity.java
package develop.y.zhzl.main;

import android.content.Context;
import android.graphics.PixelFormat;
import android.os.Bundle;
import android.view.View;
import android.view.WindowManager;

import develop.y.zhzl.R;
import framework.base.BaseActivity;

public abstract class DarkViewActivity extends BaseActivity {

    private WindowManager windowManager = null;
    private View darkView = null;
    private boolean isAddedView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        isAddedView = false;
        if (!getThemeType()) {
            initNightView();
            darkView.setBackgroundResource(R.color.night_mask);
        }
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        if (isAddedView) {
            windowManager.removeViewImmediate(darkView);
            windowManager = null;
            darkView = null;
        }
    }

    protected void initNightView() {
        if (isAddedView) {
            return;
        }
        WindowManager.LayoutParams layoutParams = new WindowManager.LayoutParams(
                WindowManager.LayoutParams.TYPE_APPLICATION,
                WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE | WindowManager.LayoutParams.FLAG_NOT_FOCUSABLE,
                PixelFormat.TRANSPARENT);
        windowManager = (WindowManager) getSystemService(Context.WINDOW_SERVICE);
        darkView = new View(this);
        windowManager.addView(darkView, layoutParams);
        isAddedView = true;
    }
}

File: app/src/main/java/develop/y/zhzl/main/widget/DarkViewActivity.kt
package develop.y.zhzl.main.widget

import android.content.Context
import android.graphics.PixelFormat
import android.os.Bundle
import android.view.View
import android.view.WindowManager

import develop.y.zhzl.R
import framework.base.BaseActivity

abstract class DarkViewActivity : BaseActivity() {

    private var mWindowManager: WindowManager? = null
    private var darkView: View? = null
    private var isAddedView: Boolean = false

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        isAddedView = false
        if (!themeType) {
            initNightView()
            darkView!!.setBackgroundResource(R.color.night_mask)
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        if (isAddedView) {
            mWindowManager!!.removeViewImmediate(darkView)
            mWindowManager = null
            darkView = null
        }
    }

    protected fun initNightView() {
        if (isAddedView) {
            return
        }
        val layoutParams = WindowManager.LayoutParams(
                WindowManager.LayoutParams.TYPE_APPLICATION,
                WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE or WindowManager.LayoutParams.FLAG_NOT_FOCUSABLE,
                PixelFormat.TRANSPARENT)
        mWindowManager = getSystemService(Context.WINDOW_SERVICE) as WindowManager
        darkView = View(this)
        mWindowManager!!.addView(darkView, layoutParams)
        isAddedView = true
    }
}
